[MRG+2] Update MovieWriter dpi default

heath730 opened the pull request:
Adds default `dpi=None` argument to *MovieWriter classes, and uses `figure.dpi` as the default if dpi is None.

This is related to issue #7616. @tacaswell please advise if this approach is what you had in mind for resolving the issue. 

I'm a new contributor and I wasn't exactly sure the best way to unit test a change like this. However, I created a test that calls setup with dpi as the default, and makes sure that dpi is set to the `figure.dpi`. I did not add a change log entry because it is such a small change, please let me know if I should.

tacaswell commented on lib/matplotlib/animation.py:
```diff
@@ -165,7 +165,7 @@ def setup(self, fig, outfile, dpi):
         dpi: int
             The DPI (or resolution) for the file.  This controls the size
-            in pixels of the resulting movie file.
+            in pixels of the resulting movie file. Default is figure.dpi.
```
The docstring on this needs `, optional` added

tacaswell replied:
probably on all of the docstrings actually.

heath730 commented:
@tacaswell I just updated the docstrings.

I just noticed that in Animate.save() has dpi=None default to rcParam['savefig.dpi'], and only if this is 'figure' it is set to figure.dpi. Should I use that behavior instead of setting None to figure.dpi?

tacaswell commented:
On one hand what you have here is simpler and does not depend on global state, on the other hand reaching out to `rcParams['savefig.dpi']` is more consistent with `savefig` and `Animation.save`.

I am leaning towards not defaulting to `rcParams['savefig.dpi']`.  This is a lower level interface (most users should be using `Animation.save` which is why this was a required argument before) and is being used by sophisticated users are are rolling their own version of `save`.  Coupling into our global state starts to be an annoying detail to work around rather than a helpful default.  It would not take much to convince me I am wrong on this and it should look at `'savefig.dpi'`.

dopplershift commented on lib/matplotlib/animation.py:
```diff
@@ -292,12 +292,14 @@ def setup(self, fig, outfile, dpi):
+        dpi : int, optional
             The DPI (or resolution) for the file.  This controls the size
-            in pixels of the resulting movie file.
+            in pixels of the resulting movie file. Default is figure.dpi.
```
Here too?

dopplershift commented on lib/matplotlib/animation.py:
```diff
@@ -155,17 +155,17 @@ class AbstractMovieWriter(six.with_metaclass(abc.ABCMeta)):
+        dpi: int, optional
             The DPI (or resolution) for the file.  This controls the size
-            in pixels of the resulting movie file.
+            in pixels of the resulting movie file. Default is figure.dpi.
```
Should figure.dpi be ``fig.dpi`` since fig is the name of the argument?

dopplershift commented on lib/matplotlib/animation.py:
```diff
@@ -413,9 +416,10 @@ def setup(self, fig, outfile, dpi, frame_prefix='_tmp', clear_temp=True):
+        dpi : number, optional
             The dpi of the output file. This, with the figure size,
             controls the size in pixels of the resulting movie file.
+            Default is figure.dpi.
```
And here?

heath730 commented:
@tacaswell @dopplershift Thanks, I updated in (c2aac89). Let me know if should rebase this on master/squash some of these commits to something cleaner.

dopplershift commented:
If it's not a lot of effort, I'd prefer to see this squashed down a bit.

heath730 commented:
@dopplershift I squashed all the lints/nits into the original commit so it looks like it never happened. I also rebased on the current master because I was a few commits behind.

dopplershift commented:
Thanks @heath730 !
